code/game/machinery/buttons.dm

```diff
@@ -5,11 +5,60 @@
 	desc = "A remote control switch for a mass driver."
 	var/id = null
 	var/active = 0
+	var/closed = 0
+	var/secure = 0
 	anchored = 1.0

```

Comment:
Is the variable secure really necessary? And isn't the variable closed only necessary in the secure button? I think this could be done with a simple override like:
`/obj/machinery/driver_button/secure/attack_hand(mob/user as mob)
	if(closed)
		src.add_fingerprint(usr)
		var/obj/machinery/driver_button/secure/S = src
		S.toggle_open(user)
		return
	..()`